dex/pair/src/safe_price_view.rs

```diff
@@ -466,14 +476,15 @@ pub trait SafePriceViewModule:
             second_token_reserve_accumulated,
             weight_accumulated,
             recording_round: search_round,
+            lp_supply_accumulated,
         }
     }
 
-    fn compute_weighted_reserves(
+    fn compute_weighted_amounts(
         &self,
         first_price_observation: &PriceObservation<Self::Api>,
         last_price_observation: &PriceObservation<Self::Api>,
-    ) -> (BigUint, BigUint) {
+    ) -> (BigUint, BigUint, BigUint) {
```

Comment:
Really bad practice to return tuples, especially if all values have the same type. Use a struct instead.

Reply:
Updated.